src/seedu/addressbook/data/person/Block.java

```diff
@@ -0,0 +1,34 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a Person's block address in the address book.
+ * Guarantees: immutable;
+ */
+
+public class Block {
+    //public final String value;
+    private boolean isPrivate;
+    public final String value;
+
+    public Block(String blockAddress, boolean isPrivate) {
```

Comment:
Do you think we can improve the classes more if we were to do one more validation step here?

Reply:
If more validations are done inside the classes, then the validation of the whole address format should also include the validation inside the classes. Hence, classes (Block, Street, etc.) need to be created, or the address string need to be split before the validations by each classes. I am a bit not sure whether it would conform to SLAP requirement. I wanted to split address string first and form the classes first then check within the class, but soon realised that the validation function for address format would also serve to form the sub-classes. I didn't know how to separate the processes so I did not attempt to validate within the class